avaframe/tests/test_com1DFA.py

```diff
@@ -1,112 +1,211 @@
+    demTest['Nz'] = np.asarray([[0., 0., 0., 0., 0.],
+                                     [0., 0., 0., 0., 0.],
+                                     [0., 0., 0., 0., 0.],
+                                     [0., 0., 0., 0., 0.],
+                                     [0., 0., 0., 0., 0.]]) + 25.
+    # setup neighbour grid
+    headerNeighbourGrid = IOf.cASCheader()
+    headerNeighbourGrid.cellsize = 0.5
+    headerNeighbourGrid.ncols = 10
+    headerNeighbourGrid.nrows = 10
+    headerNeighbourGrid.xllcenter = 0
+    headerNeighbourGrid.yllcenter = 0
+    demTest['headerNeighbourGrid'] = headerNeighbourGrid
+    areaCell = 1 / np.cos(np.deg2rad(45))
+    demTest['areaRaster'] = np.zeros((5, 5)) + areaCell
+    demTest['rasterData'] = demData
+
+    # call function to be tested
+    demOri, dem = com1DFA.initializeMesh(cfg['GENERAL'], demOri, num)
+
+    assert dem['header'].xllcenter == demTest['header'].xllcenter
+    assert dem['header'].yllcenter == demTest['header'].yllcenter
+    assert dem['header'].ncols == demTest['header'].ncols
+    assert dem['header'].nrows == demTest['header'].nrows
+    assert dem['header'].cellsize == demTest['header'].cellsize
+    assert dem['header'].yllcenter == demTest['header'].yllcenter
+    assert np.array_equal(dem['rasterData'][0:4, 0:4], demTest['rasterData'][0:4, 0:4])
+    assert np.all(np.isnan(dem['rasterData'][0:5,4]))
+    assert abs(dem['Nx'][2,2]) == abs(dem['Nz'][2,2])
+    assert np.isclose(dem['areaRaster'][2,2], demTest['areaRaster'][2,2])
+    assert dem['headerNeighbourGrid'].xllcenter == demTest['headerNeighbourGrid'].xllcenter
+    assert dem['headerNeighbourGrid'].yllcenter == demTest['headerNeighbourGrid'].yllcenter
+    assert dem['headerNeighbourGrid'].ncols == demTest['headerNeighbourGrid'].ncols
+    assert dem['headerNeighbourGrid'].nrows == demTest['headerNeighbourGrid'].nrows
+    assert dem['headerNeighbourGrid'].cellsize == demTest['headerNeighbourGrid'].cellsize
+    assert dem['headerNeighbourGrid'].yllcenter == demTest['headerNeighbourGrid'].yllcenter
+
+
+def test_polygon2Raster():
+    """ test if polygon is converted to raster properly """
+
+    # setup required inputs
+    demHeader = IOf.cASCheader()
+    demHeader.cellsize = 1
+    demHeader.ncols = 10
+    demHeader.nrows = 10
+    demHeader.xllcenter = 0
+    demHeader.yllcenter = 0
+
+    Line = {'x': np.asarray([0, 1., 0.989, 0., 0.]), 'y': np.asarray([0., 0., 0.989, 1., 0.0])}
+    radius = 0.0001
+    th = 1.2
+
+    # call function to be tested
+    Mask = com1DFA.polygon2Raster(demHeader, Line, radius, th=th)
+
+    # setup test output
+    maskTest = np.zeros((10, 10))
+    maskTest[0, 0:2] = 1.2
+    maskTest[1, 0] = 1.2
+
+    # call function to be tested
+    Mask2 = com1DFA.polygon2Raster(demHeader, Line, 0.1, th=th)
+    maskTest2 = maskTest.copy()
+    maskTest2[1, 1] = 1.2
+
+    assert np.array_equal(maskTest, Mask)
+    assert np.array_equal(maskTest2, Mask2)
```

Comment:
maybe add a case where there are two polygons that overlap and check that we get the good warning or error...

Reply:
this is for prepareArea then, next PR